design-documents/checkout/alternative-checkout-flow.md

```diff
@@ -0,0 +1,50 @@
+# Alternative checkout flow
+
+In scope of work on GraphQL and storefront APIs we have an opportunity to improve design and features of storefront checkout.
+
+The purpose of this document is to discuss possible alternatives to current Magento checkout flow. 
+
+  1. When Quote is created?
+     * For physical products on Review & Payments step
+     * For virtual products billing address has to be entered first
+  1. Cart properties:
+     * Line Items:
+       * SKU
+       * Selected options (custom/configurable)
+       * Quantity
+       * Regular price
+       * Price
+  4. Applicable cart rules calculator arguments
+     * Cart
+     * Addresses
+     * Date & Time (admin preview)
+  2. Quote factory arguments: Quote
+     * !Line Items
+     * Dimensions (weight & size) (based on LineItems)
+     * Shipping (requited for physical products)
+       * Addresses
+       * Selected Shipping Methods
+     * Billing address (required for virtual products). How to deal with multi-payment?
```

Comment:
let's reverse this question, can multiple quotes have the same payment method and billing address? In my opinion, this is a more relevant case for e-commerce. Definitely, for processing and querying, each order will have its own billing address but for creation argument, I would suggest to keep it as a single argument.